core/embed/rust/src/ui/component/model.rs

```diff
@@ -0,0 +1,67 @@
+use core::convert::TryInto;
+
+use crate::{error, ui::geometry::Point};
+
+#[cfg(feature = "model_t1")]
+pub use super::model_t1::constants;
+#[cfg(feature = "model_t1")]
+pub use super::model_t1::theme;
+
+#[cfg(feature = "model_tt")]
+pub use super::model_tt::constants;
+#[cfg(feature = "model_tt")]
+pub use super::model_tt::theme;
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum T1Button {
+    Left,
+    Right,
+}
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum T1HidEvent {
+    ButtonPressed(T1Button),
+    ButtonReleased(T1Button),
+}
+
+impl T1HidEvent {
+    pub fn new(event: u32, button: u32, _unused: u32) -> Result<Self, error::Error> {
+        let button = match button {
+            0 => T1Button::Left,
```

Comment:
would be nice if we could share the constants between C, Rust, and Micropython.

but they're `#define`s so bindgen won't import them, right?
wondering if it's worth exporting them as consts.

Reply:
was also thinking about how to do this, would be nice for e.g. display sizes which are now defined separately too

Reply:
> but they're #defines so bindgen won't import them, right?

I think it should.

Reply:
I've verified that it is possible to import #defines with bindgen. However this change would IMO require some refactoring of the C code so I'd like to leave it to another PR.